lib/src/components/NxVulnerabilityCounter/NxVulnerabilityCounter.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import classnames from 'classnames';
+
+import { Props, propTypes } from './types';
+export { Props } from './types';
+
+import './NxVulnerabilityCounter.scss';
+
+export default function NxVulnerabilityCounter(props: Props) {
+  const {
+    criticalCount,
+    highCount,
+    mediumCount,
+    lowCount,
+    noneCount,
+    unspecifiedCount,
+    layout,
+    className,
+    ...attrs
+  } = props;
+
+  const layoutClasses = classnames('nx-vulnerability-counter-container', className, {
+    [`nx-vulnerability-counter-container--${layout}`]: layout
+  });
+
+  if (
+    typeof criticalCount !== 'number' &&
+    typeof highCount !== 'number' &&
+    typeof mediumCount !== 'number' &&
+    typeof lowCount !== 'number' &&
+    typeof noneCount !== 'number' &&
+    typeof unspecifiedCount !== 'number'
+  ) {
+    console.warn(
+        'No counts have been provided and so nothing will be rendered.'
+    );
+    return null;
+  }
+
+  return (
+    <dl className={layoutClasses} {...attrs}>
+      {typeof criticalCount === 'number' && (
```

Comment:
Edge case: `NaN` has the type `number`. So a vulnerability counter can be rendered with `NaN`. Should this be allowed?
![image](https://github.com/user-attachments/assets/e837407e-994b-495a-a33d-bfe59cf147fc)

Reply:
This is not the responsibility of this component but the consumer, same with NxSmallThreatCounter.